tests/FakeItEasy.Specs/ThenSpecs.cs

```diff
@@ -0,0 +1,233 @@
+                .x(() => result2.Should().Be(2));
+
+            "And the third call returns 2"
+                .x(() => result3.Should().Be(2));
+        }
+
+        [Scenario]
+        public void AnyCallWithInvokesAndThrows(
+            IFoo foo,
+            Action action1,
+            Action action2,
+            Exception exception)
+        {
+            "Given a fake"
+                .x(() => foo = A.Fake<IFoo>());
+
+            "And two delegates"
+                .x(() =>
+                {
+                    action1 = A.Fake<Action>();
+                    action2 = A.Fake<Action>();
+                });
+
+            "When any method is configured to invoke a delegate once, then another delegate twice, then throw an exception"
+                .x(() =>
+                    A.CallTo(foo).Invokes(action1).DoesNothing().Once()
+                        .Then.Invokes(action2).DoesNothing().Twice()
+                        .Then.Throws<InvalidOperationException>());
+
+            "And 4 calls are made on the fake"
+                .x(() =>
+                {
+                    foo.Bar();
+                    foo.Bar(1);
+                    foo.Bar(2);
+                    exception = Record.Exception(() => foo.Bar(3));
+                });
+
+            "Then the first delegate is called once"
+                .x(() => A.CallTo(() => action1()).MustHaveHappened(Repeated.Exactly.Once));
+
+            "And the second delegate is called twice"
+                .x(() => A.CallTo(() => action2()).MustHaveHappened(Repeated.Exactly.Twice));
+
+            "And the fourth call throws an exception"
+                .x(() => exception.Should().BeAnExceptionOfType<InvalidOperationException>());
+        }
+
+        [Scenario]
+        public void NonVoidMethodWithThenAndOverride1(
+            IFoo foo,
+            int result1,
+            int result2,
+            int result3)
+        {
+            "Given a fake"
+                .x(() => foo = A.Fake<IFoo>());
+
+            "When a non-void method is configured to return 1"
+                .x(() => A.CallTo(() => foo.Baz()).Returns(1));
+
+            "And the configuration is overridden to return 2 once then 3 once"
+                .x(() => A.CallTo(() => foo.Baz()).Returns(2).Once()
+                    .Then.Returns(3).Once());
+
+            "And the method is called 3 times"
+                .x(() =>
+                {
+                    result1 = foo.Baz();
+                    result2 = foo.Baz();
+                    result3 = foo.Baz();
+                });
+
+            "Then the first call returns 2"
+                .x(() => result1.Should().Be(2));
+
+            "And the second call returns 3"
+                .x(() => result2.Should().Be(3));
+
+            "And the third call returns 1"
+                .x(() => result3.Should().Be(1));
+        }
+
+        [Scenario]
+        public void NonVoidMethodWithThenAndOverride2(

```

Comment:
An alternative to the 1/2 naming would be to distinguish based on the ordering of the configs - first with then, or second with then. Not a big deal to me, as the step descriptions make all clear.